Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,284 @@
+
+            if (personWithHighestBalance.ToString().Contains(", "))
+            {
+               ReplaceCommaWithAnd(personWithHighestBalance); 
+            }
+
+            return $"{personWithHighestBalance.ToString()} had the most money ever. ¤{highestBalanceEver}.";
+        }
+
+        private static void ReplaceCommaWithAnd(StringBuilder message)
+        {
+            var index = message.ToString().LastIndexOf(", ");
+            message.Remove(index, 2).Insert(index, " and "); 
+        }
+
+        private static decimal FindHighestBalance(string[] peopleAndBalances, StringBuilder PersonWithHighestBalance)
+        {
+            var highestBalanceEver = decimal.MinValue;
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                try
+                {
+                    var highestAmountOfPerson = decimal.Parse(currentPersonData[1..].Max());
+
+                    if (highestAmountOfPerson > highestBalanceEver)
+                    {
+                        highestBalanceEver = highestAmountOfPerson;
+                        PersonWithHighestBalance.Clear();
+
+                    }
+
+                    if (highestAmountOfPerson >= highestBalanceEver)
+                    {
+                        AddPerson(currentPersonData[0], PersonWithHighestBalance);
+                    }
+
+                }
+                catch (ArgumentNullException)
+                {
+                    Console.WriteLine($"There was a person with no balance in the file, namely : {currentPersonData[0]}");
+                    break;
+                }
+            }
+
+            return highestBalanceEver;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var HighestLossEver = decimal.MinValue;
+            var PersonWithHighestLoss = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var allAmounts = currentPersonData[1..];
+
+                // calculateLoss
+
+                if (allAmounts.Length <= 1)
+                {
+                    return InValidOutput;
+                }
+
+                for (int j = 0; j < allAmounts.Length - 1; j++)
+                {
+                    try
+                    {
+                        var amount1 = decimal.Parse(allAmounts[j], NumberStyles.Currency | NumberStyles.Number);
+                        try
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4967975)

Reply:
No idea how to do that without making the code less readable. I can make some new function out of it but then I have some 4 till 6 parameters which makes things not readable

Reply:
A single try catch would do the same, you just should catch two different exception types.

Reply:
Hmm, but if I use a single try catch  where I thought about it, I think I cannot see if amount1 or amount2 is a invalid number, so I cannot tell the user which character is invalid.. Or do I overseen something here.